core/node/da_clients/src/avail/client.rs

```diff
@@ -39,37 +138,175 @@ impl DataAvailabilityClient for AvailClient {
+                    )
+                    .send()
+                    .await
+                    .map_err(to_retriable_da_error)?;
+                status_response_text = status_response
+                    .text()
+                    .await
+                    .map_err(to_retriable_da_error)?;
+                status_response_struct =
+                    serde_json::from_str(&status_response_text).map_err(to_retriable_da_error)?;
+                if status_response_struct.submission.block_hash.is_some() {
+                    break;
+                }
+                retries += 1;
+                if retries > self.config.max_retries {
+                    return Err(to_retriable_da_error(anyhow!(
+                        "Failed to get gas relay status"
+                    )));
+                }
+            }
+            return Ok(DispatchResponse {
+                blob_id: format!(
+                    "{:x}:{}",
+                    status_response_struct.submission.block_hash.unwrap(),
+                    status_response_struct.submission.extrinsic_index.unwrap()
+                ),
+            });
+        }
         let client = WsClientBuilder::default()
-            .build(self.config.api_node_url.as_str())
+            .build(self.config.api_node_url.clone().unwrap().as_str())
             .await
             .map_err(to_non_retriable_da_error)?;
 
         let extrinsic = self
             .sdk_client
+            .as_ref()
+            .unwrap()
             .build_extrinsic(&client, data)
             .await
             .map_err(to_non_retriable_da_error)?;
 
         let block_hash = self
             .sdk_client
+            .as_ref()
+            .unwrap()
             .submit_extrinsic(&client, extrinsic.as_str())
             .await
             .map_err(to_non_retriable_da_error)?;
         let tx_id = self
             .sdk_client
+            .as_ref()
+            .unwrap()
             .get_tx_id(&client, block_hash.as_str(), extrinsic.as_str())
             .await
             .map_err(to_non_retriable_da_error)?;
-
         Ok(DispatchResponse::from(format!("{}:{}", block_hash, tx_id)))
     }
 
     async fn get_inclusion_data(
         &self,
-        _blob_id: &str,
+        blob_id: &str,
     ) -> anyhow::Result<Option<InclusionData>, DAError> {
-        // TODO: implement inclusion data retrieval
-        Ok(Some(InclusionData { data: vec![] }))
+        let (block_hash, tx_idx) = blob_id.split_once(':').ok_or_else(|| DAError {
+            error: anyhow!("Invalid blob ID format"),
+            is_retriable: false,
+        })?;
+
+        let url = format!(
+            "{}/eth/proof/{}?index={}",
+            self.config.bridge_api_url, block_hash, tx_idx
+        );
+        let mut response: reqwest::Response;
+        let mut retries = self.config.max_retries;
+        let mut response_text: String;
+        let mut bridge_api_data: BridgeAPIResponse;
+        loop {
```

Comment:
we don't need this loop with a sleep here as this logic is already present in da_dispatcher
you just need to figure out if the request to the bridge API actually failed, or if the proof is just not present there, if it is not yet present - return `Ok(None)`

Reply:
Fixed, closing.